### DISCUSSION TITLE
Making the gui a separate programme?

### DISCUSSION THREAD

**Discussion by MatteoLacki** [vdemichev/DiaNN#179] [Ideas]
Hello,

I have noticed that the GUI you provided is very much independent of the rest of the programme and it seems to be simply running command-line scripts. This is so nicely done, thanks. What I like about it the most, is that one does not have to run DiaNN with it, but can supply any .bat file and the arguments of the GUI will be passed on.

What we are working on is a system for distributed computations using, among others, DiaNN.
We have a cluster of linux nodes where DiaNN is installed and we can schedule commands on these computers.
This is where my question comes into game: would it be a lot of work for you to separate the GUI from all of the rest of the code and make it work on more than windows? This would be grand, because this would, in principle, allow for people to write their plugins sending the command for running DiaNN somewhere else, perhaps into the wonderful world of Linux :)

Now, in my lab people are using a set of OS-es, some have Macs, I have a linux, most have windows.
Would it be a lot of work for you guys to make the gui alone more portable and also so that it wouldn't be necessarily boundled with all the other code?

Best wishes,

Matteo

**Comment by vdemichev** [comment]
Hi Matteo,

It's indeed possible to implement this. Thank you for the suggestion!
The GUI does work on Linux under Wine. Making it work on Linux natively will be difficult though. But I guess using Wine should not really be a problem?

Best,
Vadim

**Reply by MatteoLacki** [reply]
No, I don't think that's no problem ;)
But MacOS might be?
